lib/dor/services/client/marcxml.rb

```diff
@@ -21,6 +23,39 @@ def catkey(barcode:)
 
           raise_exception_based_on_response!(resp)
         end
+
+        # Gets MARCXML corresponding to a barcode or catkey
+        # @param barcode [String] required string representing a barcode
+        # @param catkey [String] required string representing a catkey
+        # @raise [NotFoundResponse] when the response is a 500 with "Record not found in Symphony"
+        # @raise [UnexpectedResponse] on an unsuccessful response from the server
+        # @return [String] MARCXML
+        def marcxml(barcode: nil, catkey: nil)
+          check_args(barcode, catkey)
+
+          resp = connection.get do |req|
+            req.url "#{api_version}/catalog/marcxml"
+            req.params['barcode'] = barcode unless barcode.nil?
+            req.params['catkey'] = catkey unless catkey.nil?
+          end
+
+          raise_exception_based_on_marcxml_response!(resp)
+
+          resp.body
+        end
+
+        private
+
+        def check_args(barcode, catkey)
+          raise ArgumentError, 'Barcode or catkey must be provided' if barcode.nil? && catkey.nil?
+          raise ArgumentError, 'Both barcode and catkey may not be provided' if !barcode.nil? && !catkey.nil?
+        end
+
+        def raise_exception_based_on_marcxml_response!(resp)
+          raise NotFoundResponse if !resp.success? && resp.body.match?(/Record not found in Symphony/)
```

Comment:
Would you consider adding an informative message about Symphony in the `NotFoundResponse` so we know why it's raising?

Reply:
Added.